### PR TITLE
Observation dependent variables interfaces

### PR DESCRIPTION
**Pull request by SamFayolle**

### CODE REVIEW

File: src/simulation/estimation_setup/observationOutput.cpp [hunk 1]
```diff
@@ -22,10 +22,10 @@ void checkObservationDependentVariableEnvironment(
         const SystemOfBodies& bodies,
         const std::shared_ptr< StationAngleObservationDependentVariableSettings > variableSettings )
 {
-    if( isObservationDependentVariableGroundStationProperty( variableSettings ) && variableSettings->isLinkEndDefined_ )
+    if( isObservationDependentVariableGroundStationProperty( variableSettings->variableType_ ) && variableSettings->isLinkEndDefined_ )
```
Comment (DominicDirkx) [inline src/simulation/estimation_setup/observationOutput.cpp:25]: Should this function (checkObservationDependentVariableEnvironment) not be extended for other dependent variables? Right now it only checks the station angle variables

File: include/tudat/simulation/estimation_setup/observationOutputSettings.h [hunk 2]
```diff
@@ -32,22 +34,117 @@ enum ObservationDependentVariables
...
+                                    const bool revertedLinksAllowed = true )
+    {
+        bool isCompatible = true;
+        if ( variableType_ != otherSettings->variableType_ )
+        {
+            isCompatible = false;
+        }
+        else
+        {
+            bool directLinksMatch =
```
Comment (DominicDirkx) [inline include/tudat/simulation/estimation_setup/observationOutputSettings.h:119]: could you add some comments to what this does or (even better) split it up into separate bools to make it explicit what is being checked here exactly (same for the revertedLinksMatch)?

File: include/tudat/simulation/estimation_setup/observationOutputSettings.h [hunk 3]
```diff
@@ -32,22 +34,117 @@ enum ObservationDependentVariables
...
+
+            bool revertedLinksMatch =
+                    ( ( linkEndType_ == otherSettings->originatingLinkEndType_ || linkEndType_ == unidentified_link_end || otherSettings->originatingLinkEndType_ == unidentified_link_end )
+                    && ( ( linkEndId_ == otherSettings->originatingLinkEndId_ || linkEndId_ == LinkEndId( "", "" ) || otherSettings->originatingLinkEndId_ == LinkEndId( "", "" ) ) )
+                    && ( ( originatingLinkEndType_ == otherSettings->linkEndType_ ||
+                    originatingLinkEndType_ == unidentified_link_end || otherSettings->linkEndType_ == unidentified_link_end ) )
+                    && ( ( originatingLinkEndId_ == otherSettings->linkEndId_ ||
+                    originatingLinkEndId_ == LinkEndId( "", "" ) || otherSettings->linkEndId_ == LinkEndId( "", "" ) ) ) );
+
+            if ( !directLinksMatch && ( !revertedLinksAllowed || !revertedLinksMatch ) )
```
Comment (DominicDirkx) [inline include/tudat/simulation/estimation_setup/observationOutputSettings.h:135]: This is a cool way to check whether the order of the links is relevant (right?)

Comment (SamFayolle) [reply include/tudat/simulation/estimation_setup/observationOutputSettings.h:135]: yes exactly, to give more flexibility regarding the input arguments order when defining the variable settings  (for some dependent variables, it doesn't matter if the user gets the link end order wrong). This also prevents the dependent variables being created and computed twice if the link "direction" does not matter.

File: src/simulation/estimation_setup/observationOutput.cpp (hunk 1 above)
Comment (SamFayolle) [reply src/simulation/estimation_setup/observationOutput.cpp:25]: good point, I didn't extend this function indeed

**Comment by SamFayolle** [comment]
The dependent variable filtering option is now implemented. @DominicDirkx  I have not created a similar dependent variable-based observation set **splitter**, because I could not really identify an immediate use case for this. But just let me know if you think otherwise :)

**Comment by DominicDirkx** [comment]
Cool, thanks! I don't really see (for now) much of a use of a set-splitter, let's leave it for now.

I see you commented out some of the test cases in unitTestObservationsProcessing.cpp, I assume to speed up your local compilation and run (?) Except for adding this back, is there anything that you want to add before we can merge this?

**Comment by SamFayolle** [comment]
> I see you commented out some of the test cases in unitTestObservationsProcessing.cpp
Yes, I realised that too, sorry! They will be back with the next commit

> Except for adding this back, is there anything that you want to add before we can merge this?
nothing feature-wise, but I thought I spent some time adding some proper comments this afternoon, because that's really missing

**Comment by SamFayolle** [comment]
For now, I only added comments to the new dependent variables stuff but I realised that the whole observations.h file could use a lot more comments in general (also for the new parser/filter/splitter functionalities). I don't think that it is necessary for this PR though(?) but something to keep in mind

**Comment by DominicDirkx** [comment]
Merged :)

**Pull request merged**